rosbag2_compression/src/rosbag2_compression/compression_factory_impl.hpp

```diff
@@ -15,11 +15,34 @@
 #ifndef ROSBAG2_COMPRESSION__COMPRESSION_FACTORY_IMPL_HPP_
 #define ROSBAG2_COMPRESSION__COMPRESSION_FACTORY_IMPL_HPP_
 
+#include <algorithm>
 #include <memory>
 #include <stdexcept>
 #include <string>
 
+#include "logging.hpp"
 #include "rosbag2_compression/compression_factory.hpp"
+#include "rosbag2_compression/zstd_compressor.hpp"
+#include "rosbag2_compression/zstd_decompressor.hpp"
+
+namespace
+{
+
+constexpr const char kCompressionFormatZstd[] = "zstd";
```

Comment:
Why anonymous? If not, then others can use and in the tests as well. Or another question, why not use the identifier provided by the zstd compressor?

Reply:
I'd need to make `get_compression_identifier()` static which I'd prefer to do in another PR. But yes that would make this less repetitive.

Reply:
I don't think you can make `BaseCompressorInterface::get_compression_identifier` static since it needs to be implemented by each supported compression. You can add a static function to `ZstdCompressor`